recipe/conda_build_config.yaml

```diff
@@ -0,0 +1,3 @@
+c_compiler_version:
+  - 18.1.5                                 # [osx and arm64]
```

Comment:
18.1.5 is not receiving any updates in our infrastructure. If you need to stay below 18.1.6, better to just use clang 17.

However, the clang release notes you found (well done!) point out a much better solution:
> The extension can be turned off by the option `-fno-define-target-os-macros` as a workaround.

Reply:
So you should be able to do something like
```
export CFLAGS="$CFLAGS -fno-define-target-os-macros"
```
in the build script

Reply:
How can I proceed with the build script? Should I make a build.sh specifically for osx?

Reply:
Nevermind, I took a look into some other recipe in conda-forge and I can add it to the meta.yaml